src/cloud-resources/src/crd/materialize.rs

```diff
@@ -108,6 +108,12 @@ pub mod v1alpha1 {
         // generation rollout is automatically triggered.
         #[serde(default = "Uuid::new_v4")]
         pub request_rollout: Uuid,
+        // If force_promote is set to the same value as request_rollout, the
+        // current rollout will skip waiting for clusters in the new
+        // generation to rehydrate before promoting the new environmentd to
+        // leader.
+        #[serde(default)]
+        pub force_promote: Uuid,
```

Comment:
Does this need to be an `Option<Uuid>` to handle already existing CRs? What does K8S do in cases where a new CRD definition is incompatible with the old stored data?

The `#[serde(default)]` handles missing data on the controller side, but does K8S even allow you to upgrade the CRD in this way?

Reply:
as far as i can tell from testing, this works as you would want it to